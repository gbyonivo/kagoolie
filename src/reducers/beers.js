import { FETCH_BEERS, FINISHED_FETCHING_BEERS, ERROR_FETCHING_BEERS } from '../constants/actionTypes';

const initialState = {
  beers: [],
  errors: [],
  isFetchingBeers: true
};

const ACTION_HANDLERS = {
  [FETCH_BEERS]: state => ({
    ...state,
    isFetchingBeers: true
  }),
  [FINISHED_FETCHING_BEERS]: (state, { beers }) => ({
    ...state,
    beers,
    isFetchingBeers: false
  }),
  [ERROR_FETCHING_BEERS]: (state, { error }) => ({
    ...state,
    errors: [...state.errors, error],
    isFetchingBeers: false
  })
};

export default (state = initialState, { payload, type }) => {
  const handler = ACTION_HANDLERS[type];
  return handler ? handler(state, payload) : state;
};