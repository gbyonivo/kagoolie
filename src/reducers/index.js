import { combineReducers } from 'redux';
import beersReducer from './beers';

export default () => combineReducers({
  beersReducer
});