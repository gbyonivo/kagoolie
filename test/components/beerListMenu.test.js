import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import reducers from '../__testData__/store';
import combinedReducer from '../../src/reducers';
import BeersListMenu from '../../src/components/beersListMenu';

describe('BeersListMenu component', () => {
  it('renders correctly', () => {
    const store = createStore(combinedReducer(), reducers);
    const actual = renderer.create(<Provider store={store}>
      <BeersListMenu />
    </Provider>);
    expect(actual).toMatchSnapshot();
  });
});