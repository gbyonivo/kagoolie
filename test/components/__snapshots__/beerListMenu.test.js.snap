// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeersListMenu component renders correctly 1`] = `
<div
  className="beersListMenu"
>
  <h2>
    BEERS
  </h2>
  <ul>
    <li
      className="beerListMenuItem isCollapsed"
    >
      <h4
        onClick={[Function]}
      >
        jupiter
      </h4>
      <div>
        <div
          className="tagline"
        >
          <em>
            Tagline: 
          </em>
          <span />
        </div>
        <div
          className="date"
        >
          <em>
            First Brewed: 
          </em>
          <span />
        </div>
        <div
          className="description"
        >
          descriptive
        </div>
        <div
          className="ingredientsWrapper"
        >
          <h6>
            <em>
              Ingredients
            </em>
          </h6>
          <div
            className="ingredients"
          >
            <div
              className="ingredient"
            >
              malt
            </div>
            <div
              className="ingredientDetails"
            >
              <span>
                malting
                <span
                  className="ingredientAmount"
                >
                   undefined
                  undefined, 
                </span>
              </span>
            </div>
          </div>
          <div
            className="ingredients"
          >
            <div
              className="ingredient"
            >
              yeast
            </div>
            <div
              className="ingredientDetails"
            >
              firmino
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      className="beerListMenuItem isCollapsed"
    >
      <h4
        onClick={[Function]}
      >
        pluto
      </h4>
      <div>
        <div
          className="tagline"
        >
          <em>
            Tagline: 
          </em>
          <span />
        </div>
        <div
          className="date"
        >
          <em>
            First Brewed: 
          </em>
          <span />
        </div>
        <div
          className="description"
        >
          description
        </div>
        <div
          className="ingredientsWrapper"
        >
          <h6>
            <em>
              Ingredients
            </em>
          </h6>
          <div
            className="ingredients"
          >
            <div
              className="ingredient"
            >
              malt
            </div>
            <div
              className="ingredientDetails"
            >
              <span>
                malter
                <span
                  className="ingredientAmount"
                >
                   undefined
                  undefined, 
                </span>
              </span>
            </div>
          </div>
          <div
            className="ingredients"
          >
            <div
              className="ingredient"
            >
              yeast
            </div>
            <div
              className="ingredientDetails"
            >
              salah
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
